js/ui/appDisplay.js

```diff
@@ -1602,9 +1602,9 @@ var AppSearchProvider = new Lang.Class({
             let hasA = a === EOS_APP_CENTER_ID || IconGridLayout.layout.hasIcon(a);
             let hasB = b === EOS_APP_CENTER_ID || IconGridLayout.layout.hasIcon(b);
 
-            if (hasA)
+            if (hasA && !hasB)
                 return -1;
-            if (hasB)
+            if (hasB && !hasA)
```

Comment:
This can be simplified as `return hasA - hasB;`

Reply:
Do you really think treating the booleans as integers and doing arithmetic on them is clearer?

Reply:
I am totally biased because C, so yes I do. And I only see benefits there:

 * Less lines of code
 * Sorting functions actually looks like a sorting function, rather than a table test
 * Three less branches to mentally track

Reply:
Okay

Reply:
(testing the new patch now)